sklearn/linear_model/least_angle.py

```diff
@@ -642,13 +622,6 @@ def fit(self, X, y, Xy=None):
 
         n_targets = y.shape[1]
 
-        alpha = getattr(self, 'alpha', 0.)
-        if hasattr(self, 'n_nonzero_coefs'):
-            alpha = 0.  # n_nonzero_coefs parametrization takes priority
-            max_iter = self.n_nonzero_coefs
-        else:
-            max_iter = self.max_iter
```

Comment:
this has been moved to Lars.fit to avoid the magic of hacking instance attributes